tests/manage/cluster/test_add_mds_to_cluster.py

```diff
@@ -7,88 +7,76 @@
     """
     Setting up the environment for the test
     """
-    global CEPHFS, CEPH_OBJ
-    CEPHFS = ocp.OCP(kind=constants.CEPHFILESYSTEM)
-    self.fs_data = CEPHFS.get(defaults.CEPHFILESYSTEM_NAME)
-    self.fs_name = self.fs_data['metadata']['name']
-    CEPH_OBJ = OCS(**self.fs_data)
-
+    def finalizer():
+        fs_data = cephfs.get(defaults.CEPHFILESYSTEM_NAME)
+        fs_data['spec']['metadataServer']['activeCount'] = (
+            original_active_count
+        )
+        ceph_obj.apply(**fs_data)
 
-def teardown(self):
-    """
-    Tearing down the environment
-    """
-    self.fs_data = CEPHFS.get(defaults.CEPHFILESYSTEM_NAME)
-    self.fs_data['spec']['metadataServer']['activeCount'] = (
-        self.original_active_count
-    )
-    CEPH_OBJ.apply(**self.fs_data)
-    assert verify_fs_exist(self.original_active_count * 2)
+    request.addfinalizer(finalizer)
+    cephfs = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+    fs_data = cephfs.get(defaults.CEPHFILESYSTEM_NAME)
+    original_active_count = fs_data.get('spec').get('metadataServer').get('activeCount')
+    ceph_obj = OCS(**fs_data)
+    return original_active_count, fs_data, ceph_obj
 
 
-def verify_fs_exist(pod_count):
+def mds_active_count():
     """
-    Verifying if a ceph FS exist
+    Return Active MDS Count
     """
-    assert POD.wait_for_resource(
-        condition='Running', selector='app=rook-ceph-mds',
-        resource_count=pod_count
-    )
-    pods = POD.get(selector='app=rook-ceph-mds')['items']
-    if len(pods) == pod_count:
-        return True
-    return False
+    pod = ocp.OCP(kind=constants.POD,
+                  namespace=config.ENV_DATA['cluster_namespace']
+                  )
+    pods = pod.get(selector='app=rook-ceph-mds')['items']
+    cephfs = ocp.OCP(kind=constants.CEPHFILESYSTEM)
+    fs_data = cephfs.get(defaults.CEPHFILESYSTEM_NAME)
+    new_active_count = fs_data.get('spec').get('metadataServer').get('activeCount')
+    return new_active_count, pods
 
 
+@ignore_leftovers
 @bugzilla('1752286')
 @tier1
-@pytest.mark.usefixtures(
-    test_fixture.__name__,
-)
 class TestCephFilesystemCreation(ManageTest):
     """
     Testing creation of Ceph FileSystem
     """
-    new_active_count = 2
-    original_active_count = 1
-    fs_name = None
-
-    def test_cephfilesystem_creation(self):
+    def test_cephfilesystem_creation(
+        self,
+        fs_setup
+    ):
         """
         Creating a Ceph Filesystem
         """
+        original_active_count, fs_data, ceph_obj = fs_setup
+        new_active_count = original_active_count + 1
 
-        self.fs_data['spec']['metadataServer']['activeCount'] = (
-            self.new_active_count
+        fs_data['spec']['metadataServer']['activeCount'] = (
+            new_active_count
         )
-        CEPH_OBJ.apply(**self.fs_data)
-        assert verify_fs_exist(self.new_active_count * 2)
+        ceph_obj.apply(**fs_data)
+        for mdss, pods in TimeoutSampler(
```

Comment:
why mdss? Isn't mds_ac better?

Reply:
mdss is better, mds -> metadata server (a noun) , mdss -> metadata servers.